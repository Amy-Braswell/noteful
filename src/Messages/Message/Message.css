
.Message {
    --message-height: 88px;
    background-color: #010d1e;
    color: #cacaca;
    height: var(--message-height);
    margin-bottom: 8px;
    max-width: 380px;
    padding: 8px 0 8px 8px;
    position: relative;
  }
  
  .Message__delete {
    --message-button-height: 32px;
    background-color: #041a3a;
    border: 0;
    color: #bdbcbc;
    float: right;
    font-size: 12px;
    height: var(--message-button-height);
    position: relative;
    top: calc((var(--message-button-height) + 32px) * -1);
    transform-origin: right bottom;
    transform: rotate(-90deg);
    width: calc(var(--message-height));
  }
  .Message__delete:hover {
    background-color: #bd0909;
    color: white;
    cursor: pointer;
    font-weight: bold;
  }
  
  .Message__title {
    font-family: var(--custom-font);
    font-size: 20px;
    margin: 0;
    max-width: 80%;
    min-height: 24px;
  }
  .Message__title a {
    color: #cacaca;
    display: block;
    text-decoration: none;
  }
  .Message__title a:hover {
    color: #FFB72F;
  }

  p.Content {
    bottom: 15px;
    position: absolute;
  }
  
  .Message__dates {
    bottom: 8px;
    color: #a7a5a5;
    font-size: 14px;
    position: absolute;
  }
  