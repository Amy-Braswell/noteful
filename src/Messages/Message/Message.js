import React from 'react'
import { Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { format } from 'date-fns'
import ApiContext from '../../ApiContext'
import config from '../../config'
import './Message.css'

export default class Message extends React.Component {

  static defaultProps ={
    onDeleteMessage: () => {},
  }
  static contextType = ApiContext


  handleClickDelete = () => {
    const messageId = this.props.id
    const options = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json'
    }}
    const url = (`${config.API_ENDPOINT}/messages/${messageId}`)
      fetch(url, options)
      .then(res => {
        if (!res.ok) {
          throw new Error('Something went wrong')
        }
        return res.json
      })
      .then(() => {
        this.context.handleDeleteMessage(messageId)
        // allow parent to perform extra behaviour
        this.props.onDeleteMessage(messageId)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  
  render() {
    const {name, id, modified} = this.props
    if (!name || !id || !modified) {
      return null
    } 
    const formatDate = format(new Date(modified), 'hh:mm a / MMMM do yyyy')
    return (
      <div className='Message'>
        <h2 className='Message__title'>
          <Link to={`/messages/${id}`}>
            {name}
          </Link>
        </h2>
        <button
          className='Message__delete'
          type='button'
          onClick={this.handleClickDelete}
        >
          <FontAwesomeIcon icon={['fa', 'trash-alt']} />
          {' '}
          remove
        </button>
        <div className='Message__dates'>
          <div className='Message__dates-modified'>
            Modified
            {' '}
            <span className='Date'>
            {formatDate}
            </span>
          </div>
        </div>
      </div>
    )   
  }
}

