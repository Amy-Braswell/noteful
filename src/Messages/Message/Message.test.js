import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Message from './Message'

describe(`Message component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    modified: new Date(2018, 12, 15),
  }

  it('renders a .Message by default', () => {
    const wrapper = shallow(<Message />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Message given props', () => {
    const wrapper = shallow(<Message {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
