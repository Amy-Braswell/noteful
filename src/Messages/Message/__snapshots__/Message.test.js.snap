// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component renders a .Message by default 1`] = `""`;

exports[`Message component renders the Message given props 1`] = `""`;
